packages/nimble-components/src/table-column/mapping/models/column-validator.ts

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable max-classes-per-file */
+import { nullableNumberConverter } from '@microsoft/fast-element';
+import { Icon } from '../../../icon-base';
+import { ColumnValidator } from '../../base/models/column-validator';
+import type { Mapping } from '../mappings';
+import { MappingIcon } from '../mappings/icon';
+
+export const mappingColumnValidityFlagNames = [
+    'invalidMappingKeyValueForType',
+    'multipleDefaultMappings',
+    'unsupportedMappingType',
+    'duplicateMappingKey',
+    'missingKeyValue'
+] as const;
+
+/**
+ * Helper to share logic between the mapping and icon column validators
+ */
+class TableColumnMappingValidationHelper {
```

Comment:
I'm using this helper class to share logic, because I can't share via inheritance. Even though there is almost complete overlap between the validation done for each column type, I can't give the two different validators a shared `ColumnValidator` base class. It needs the set of validity flag names as the type parameter, and those are not exactly the same for the two validators (icon column additionally has "invalidIconName").